xla/service/gpu/runtime/cudnn_thunk.h

```diff
@@ -43,7 +43,7 @@ class CuDnnThunk : public Thunk {
   absl::Status Initialize(const InitializeParams&) override;
   absl::Status ExecuteOnStream(const ExecuteParams&) override;
 
-  const se::dnn::DnnGraph& graph() const { return *graph_; }
+  const std::unique_ptr<se::dnn::DnnGraph>& graph() const { return graph_; }
```

Comment:
How about returning `std::shared_ptr<LazyDnnGraph>` (or something like this), and whoever gets called first (Thunk or corresponding Command) will initialize it.

Basically command emitter should steal (or share) a state from a thunk

Reply:
I just added a call to Initialize() to ExecuteOnStream() - does this look reasonable to you? Body of Initialize() is already wrapped in call_once().